docs/source/user_guide_rewrite.md

```diff
@@ -0,0 +1,448 @@
+
+Much like machine learning libraries have done for prediction, "DoWhy" is a Python library that aims to spark causal thinking and analysis. DoWhy provides a principled four-step interface for causal inference that focuses on explicitly modeling causal assumptions and validating them as much as possible. The key feature of DoWhy is its state-of-the-art refutation API that can automatically test causal assumptions for any estimation method, thus making inference more robust and accessible to non-experts. DoWhy supports estimation of the average causal effect for backdoor, frontdoor, instrumental variable and other identification methods, and estimation of the conditional effect (CATE) through an integration with the EconML library.
+
+
+*In general, it should create some motivation and excitement for why one would use this library and why it’s needed.*
+
+# Introduction
+
+*[Maybe also include: “Who is this library/user guide for?”]*
+
+DoWhy helps to perform a wide variety of causal *tasks* based on a method generally referred to as causal inference. These tasks help to give answers to questions, such as “If I change the color of my button to red, how much will this change users’ purchase decisions (effect)?”, or “Which service in my distributed system has caused the frontend to be slower than usual?”.
+
+To give readers easy access to learn about such tasks, this user guide dedicates the entire chapter [Performing Causal Tasks](#performing-causal-tasks) to explain how DoWhy can be used to perform such tasks. We categorize tasks into [Estimating Causal Effects](#estimating-causal-effects), [Root-Causing and Explaining Observed Effects](#root-causing-and-explaining-observed-effects), and [Asking and Answering What-If Questions](#asking-and-answering-what-if-questions).
+
+To perform tasks, DoWhy leverages two powerful frameworks, namely graphical causal models (GCM) and potential outcomes (PO). Depending on the task at hand, it leverages GCM, or PO, or both, or none. What’s common to all tasks though, except for unit-change and feature attribution, is that they require a *causal graph*, which is modeled after the problem domain. For that reason, this user guide starts with [Modeling Causal Relations](#modeling-causal-relations). Readers interested in [Unit-Change Attribution](#unit-change-attribution) or [Feature Attribution](#feature-attribution) can skip right to those sections.
+
+To aid the navigation within this user guide and the library further, the following flow chart can be used:
+
+*[TODO (pego): insert flow chart here, which is a* *visual navigation map for the library.]*
```

Comment:
great idea!

Reply:
I've added this now. See https://github.com/py-why/dowhy/blob/cc4b2e978d0e6de598f77ff87623ecd15475fc88/docs/source/user_guide/navigation.png (or simply the latest diff of this PR).